arbor/partition_load_balance.cpp

```diff
@@ -1,16 +1,20 @@
 #include <arbor/distributed_context.hpp>
 #include <arbor/domain_decomposition.hpp>
+#include <arbor/load_balance.hpp>
 #include <arbor/recipe.hpp>
 
-#include "hardware/node_info.hpp"
+#include "cell_group_factory.hpp"
+#include "util/maputil.hpp"
 #include "util/partition.hpp"
 #include "util/span.hpp"
 
 namespace arb {
 
-domain_decomposition partition_load_balance(const recipe& rec,
-                                            hw::node_info nd,
-                                            const distributed_context* ctx)
+domain_decomposition partition_load_balance(
+    const recipe& rec,
+    domain_info nd,
```

Comment:
I don't fell that `domain_info` is a good name, because I think of a domain as a chunk/part of the model that is running on the local resources.
A more appropriate name might make it clear that this refers to hardware resources.

Reply:
Sure — suggestions below.